### PR TITLE
auto-delete bucket

### PR DESCRIPTION
**Pull request by floridemai**
A simple implementation of a custom resource that empties the s3 bucket when the DELETE event is received from the stack. Note: if for some reason the delete operation fails there is the chance of having a partially deleted bucket. Subsequent calls for deletion won't be impacted.
I've written tests to cover the code for the most part of it but suggestions on what I might have missed are welcomed. And as always... text describing the tests...

```
 ✅  PaulDev-TestAutoDelete-Service-Default

Outputs:
PaulDev-TestAutoDelete-Service-Default.WebsiteURL = http://pauldev-testautodelete-service-sitebucket397a1860-rwt4yw1y5k0l.s3-website-eu-west-1.amazonaws.com
```
```
 ✅  PaulDev-TestAutoDelete-Service-Default: destroyed
```

### CODE REVIEW

File: src/cdk/constructs/auto-delete-bucket/auto-delete-bucket.test.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    expect(stack).toHaveResource('AWS::S3::Bucket')
+
+    expect(stack).toHaveResourceLike('AWS::Lambda::Function', {
+      Handler: 'lambda/index.handler',
+      Runtime: 'nodejs10.x',
+      Timeout: 900
+    })
+
+    expect(stack).toHaveResource('AWS::IAM::Policy')
+    //   expect(stack).toHaveResourceLike('AWS::IAM::Policy', {
```
Comment (zimny) [inline src/cdk/constructs/auto-delete-bucket/auto-delete-bucket.test.ts:30]: can we remove code if it's not used?

File: src/cdk/custom-resources/auto-delete-bucket/utils/send-response.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    )
+
+    req.once('error', (err) => (error = err))
+    req.once('close', () => (error ? reject(error) : resolve()))
+
+    req.write(responseBody)
+    req.end()
+  })
+
+  // await fetch.put(options.responseUrl, responseBody, {
```
Comment (zimny) [inline src/cdk/custom-resources/auto-delete-bucket/utils/send-response.ts:63]: can we delete it?

File: src/cdk/constructs/auto-delete-bucket/auto-delete-bucket.test.ts (hunk 1 above)
Comment (floridemai) [reply src/cdk/constructs/auto-delete-bucket/auto-delete-bucket.test.ts:30]: If it's ok with you, I'd rather keep it as I really want to have it in at some point (i'll keep an eye on it, I promise).
It highlights that there should be a better test there but the lack of support from @aws-cdk/assert stops us from doing that at this point.

Comment (zimny) [reply src/cdk/constructs/auto-delete-bucket/auto-delete-bucket.test.ts:30]: Can you add simple todo near this comment? ;)

**Pull request merged**